internal/diskutils/diskutil.go

```diff
@@ -47,245 +36,77 @@ type Command interface {
-	if err != nil {
-		return []BlockDevice{}, []BlockDevice{}, fmt.Errorf("failed to run command: %s", err)
-	}
-	if output == "" {
-		return []BlockDevice{}, []BlockDevice{}, nil
-	}
-	lDevices := BlockDeviceList{}
-	err = json.Unmarshal([]byte(output), &lDevices)
-	if err != nil {
-		return []BlockDevice{}, []BlockDevice{}, fmt.Errorf("failed to unmarshal JSON %s: %s", output, err)
-	}
-
-	for idx := range lDevices.BlockDevices {
-		// only use device if name is populated, and non-empty
-		if strings.Trim(lDevices.BlockDevices[idx].Name, " ") == "" {
-			badRows = append(badRows, lDevices.BlockDevices[idx])
-			e, err := json.Marshal(badRows)
-			m := fmt.Sprintf("Found an entry with empty name: %s.", e)
-			if err != nil {
-				m = fmt.Sprintf(m+" Failed to marshal ", err)
-			}
-			klog.Warning(m)
-			break
-		}
-
-		lDevices.BlockDevices[idx].Model = strings.Trim(lDevices.BlockDevices[idx].Model, " ")
-		lDevices.BlockDevices[idx].Vendor = strings.Trim(lDevices.BlockDevices[idx].Vendor, " ")
-		// Update device filesystem using `blkid`
-		if fs, ok := deviceFSMap[fmt.Sprintf("/dev/%s", lDevices.BlockDevices[idx].Name)]; ok {
-			lDevices.BlockDevices[idx].FSType = fs
-		}
-		blockDevices = append(blockDevices, lDevices.BlockDevices[idx])
-	}
-
-	if len(badRows) == len(lDevices.BlockDevices) && len(lDevices.BlockDevices) > 0 {
-		return []BlockDevice{}, badRows, fmt.Errorf("could not parse any of the lsblk entries")
-	}
-
-	return blockDevices, badRows, nil
-}
-
-// GetDeviceFSMap returns mapping between disks and the filesystem using blkid
-// It parses the output of `blkid -s TYPE`. Sample output format before parsing
-// `/dev/sdc: TYPE="ext4"
-// /dev/sdd: TYPE="ext2"`
-// If devices is empty, it scans all disks, otherwise only devices.
-func GetDeviceFSMap(devices []string) (map[string]string, error) {
-	m := map[string]string{}
-	args := append([]string{"-s", "TYPE"}, devices...)
-	cmd := ExecCommand.Execute("blkid", args...)
-	output, err := executeCmdWithCombinedOutput(cmd)
-	if err != nil {
-		// According to blkid man page, exit status 2 is returned
-		// if no device found.
-		if exiterr, ok := err.(*exec.ExitError); ok {
-			if exiterr.ExitCode() == 2 {
-				return map[string]string{}, nil
-			}
-		}
-		return map[string]string{}, err
-	}
-	lines := strings.Split(output, "\n")
-	for _, l := range lines {
-		if l == "" {
-			// Ignore empty line.
-			continue
-		}
-
-		values := strings.Split(l, ":")
-		if len(values) != 2 {
-			continue
-		}
-
-		fs := strings.Split(values[1], "=")
-		if len(fs) != 2 {
-			continue
-		}
-
-		m[values[0]] = strings.Trim(strings.TrimSpace(fs[1]), "\"")
-	}
-
-	return m, nil
-}
-
-func executeCmdWithCombinedOutput(cmd Command) (string, error) {
 	output, err := cmd.CombinedOutput()
 	if err != nil {
-		return string(output), err
+		return []byte{}, fmt.Errorf("failed to run command: %s", err)
 	}
-	return strings.TrimSpace(string(output)), nil
+	return output, err
```

Comment:
err is nil here, I'd return nil instead just to be more idiomatic. It's usual to either return nil (or default value when it's not a pointer) with an error and the opposite than return both variables.
```
return "this is a valid return",nil // when no error occurred
```
or 
```
return "",fmt.Errorf("this failed miserably")
```